lib/api.go

```diff
@@ -0,0 +1,63 @@
+package sebak
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"boscoin.io/sebak/lib/network"
+	"github.com/GianlucaGuarini/go-observable"
+)
+
```

Comment:
I know my approach is ugly. `T_T`
```
⚠ func (nr *NodeRunner) Ready() {
      nr.network.SetContext(nr.ctx)
      nr.network.AddHandler(nr.ctx, AddAPIHandlers(nr.storage))
      nr.network.Ready()
  }
```

```
⚠ func AddAPIHandlers(s *sebakstorage.LevelDBBackend) func(ctx context.Context, t *sebaknetwork.HTTP2Network) {
      fn := func(ctx context.Context, t *sebaknetwork.HTTP2Network) {
          //t.AddAPIHandler(ctx, "/account/{address}", GetAccountHandler).Methods("GET")
          t.AddAPIHandler("/account/{address}", GetAccountHandler(ctx, t, s)).Methods("GET")
      }
      return fn
  }
```

```
func GetAccountHandler(ctx context.Context, t *sebaknetwork.HTTP2Network, s *sebakstorage.LevelDBBackend) http.Handle
  rFunc {
      /*
          storage, ok := ctx.Value("storage").(*sebakstorage.LevelDBBackend)
          if !ok {
              panic(errors.New("storage is missing in context"))
          }
      */

      storage := s
```

```
⚠ func (t *HTTP2Network) AddAPIHandler(pattern string, handlerFunc http.HandlerFunc) (router *mux.Route) {
      //pattern := args[0].(string)
      //handler := args[1].(func(context.Context, *HTTP2Network) HandlerFunc)
      apiRouter := t.routers["api"]
      return apiRouter.HandleFunc(pattern, handlerFunc)
  }
```

Reply:
It is not ugly. elegance. 👍 
I have been suffered from circular import, hence my dirty code been written.
Your idea makes me free from circular import and refer to the storage
Thanks @anarcher

Reply:
@kfangw  I am honored that I have helped you. `:-]`